File: src/main/java/uk/edu/populationfitness/simulation/Simulation.java
package uk.edu.populationfitness.simulation;

import uk.edu.populationfitness.models.Generations;

/***
 * Implement this to run a simulation
 */
public class Simulation extends Thread {

    public Generations generations;

    final int parallel_run;

    Simulation(int parallel_run) {
        this.parallel_run = parallel_run;
    }
}

File: DotNet/PopulationFitness/PopulationFitness/Simulation/Simulation.cs
using PopulationFitness.Models;

namespace PopulationFitness.Simulation
{
    /***
     * Implement this to run a simulation
     */
    public abstract class Simulation
    {
        private Generations _generations = null;

        public Generations Generations
        {
            get
            {
                lock(this)
                {
                    return _generations;
                }
            }
            set
            {
                lock(this)
                {
                    _generations = value;
                }
            }
        }

        protected readonly int RunNumber;

        protected Simulation(int run)
        {
            RunNumber = run;
        }

        /**
         * Implement this to run the simulation
         */
        public abstract void Start();

        /**
         * Implement this to join the thread
         */
        public abstract void Join();
    }
}
